### PR TITLE
refactor ReviewListView in dashboard

### PR DESCRIPTION
**Pull request by pjstevns**
- split out the queryset filters into separate methods for easy overloading
  in subclasses, i.e. have the keyword search in product titles as well.
- cleanup the get_date_from_to_queryset method a little to make it more dry.

### CODE REVIEW

File: tests/integration/dashboard/test_reviews.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+@pytest.fixture
+def reviewlistview(rf):
+    view = views.ReviewListView()
+    view.request = rf.get('/')
+    return view
+
+
+@pytest.mark.django_db
+class TestReviewListView(object):
```
Comment (solarissmoke) [inline tests/integration/dashboard/test_reviews.py]: I think we should also be testing whether the queryset/filters that are generated by `get_queryset()` and submethods is actually correct? It ends up being quite a complex query and currently we don't appear to have any tests to check that it is correct.

Comment (sasha0) [reply tests/integration/dashboard/test_reviews.py]: Indeed.

Comment (sasha0) [reply tests/integration/dashboard/test_reviews.py]: @pjstevns would you add suggested test?

Comment (pjstevns) [reply tests/integration/dashboard/test_reviews.py]: @sasha0 I've expanded the test-coverage

**Pull request merged**